model/task/task.go

```diff
@@ -158,8 +158,23 @@ type Task struct {
 	ExecutionPlatform ExecutionPlatform `bson:"execution_platform,omitempty" json:"execution_platform,omitempty"`
 
 	// The version of the agent this task was run on.
-	AgentVersion string                 `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
-	TaskOutput   *taskoutput.TaskOutput `bson:"task_output,omitempty" json:"task_output,omitempty"`
+	AgentVersion string `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
+	// TaskOutputInfo holds the information for the interface that
+	// coordinates persistent storage of a task's output data.
+	// There are four possible scenarios:
+	//     1. The task will never have output data (e.g., display tasks)
+	//        and, therefore, the value is and always will be nil.
+	//     2. The task does not have output data yet, but can in the future
+	//        after running, and, therefore, the value is currently nil.
```

Comment:
"once it is dispatched" rather than "after running" since the value is no longer nil after it's dispatched. This seems like this case encompasses when the task is dispatched/currently running, in which case this field is not nil and it should have output data eventually, but may not have the complete output yet. I would split these two cases into scheduled (i.e. it's nil and may eventually run), dispatched/running (i.e. the field is non-nil but the output may not be done), and finished (i.e. the field is non-nil and output is finished).